### ISSUE TITLE
HeadBucketRequest doesn't provide equals method implementation

### ISSUE THREAD

**Issue by sf-mkirilov** [aws/aws-sdk-java#2073]
I expect from the following code snippet to print 'true' but it doesn't.

```java
HeadBucketRequest request1 = new HeadBucketRequest("dummy-bucket");
HeadBucketRequest request2 = new HeadBucketRequest("dummy-bucket");
System.out.println(request1.equals(request2));
```

I encountered this during unit testing of my code with mocks that respond differently to different head bucket requests.

In case this is not by design I will be glad to contribute a PR.

**Comment by debora-ito** [comment]
Marking as a feature request, if you can send us a PR we are happy to take a look.

**Comment by debora-ito** [comment]
Change released in `1.11.633`.
